controllers/k8ssandra/datacenters.go

```diff
@@ -233,17 +247,60 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 		}
 	}
 
+	if AllowUpdate(kc) {
+		dcsRequiringUpdate := make([]string, 0, len(actualDcs))
+		for _, dc := range actualDcs {
+			if dc.Status.GetConditionStatus("RequiresUpdate") == corev1.ConditionTrue { // TODO Update this to cassdcapi const when cass-operator is updatedß
+				dcsRequiringUpdate = append(dcsRequiringUpdate, dc.DatacenterName())
+			}
+		}
+
+		if len(dcsRequiringUpdate) > 0 {
+			generatedName := fmt.Sprintf("refresh-%d", time.Now().Unix())
+			internalTask := &ktaskapi.K8ssandraTask{}
+			err := r.Get(ctx, types.NamespacedName{Namespace: kc.Namespace, Name: generatedName}, internalTask)
+			// If task wasn't found, create it and if task is still running, requeue
```

Comment:
[issue] If I'm following correctly, we might requeue for a few reconciliation loops while we wait for the task to finish. However, each loop creates its own `generatedName`, which might be different because the UNIX timestamp part has a resolution of one second. With bad timing, we might end up creating multiple tasks.

Reply:
Just to confirm, manual testing confirms that the operator keeps creating refresh tasks in a loop.

```yaml
apiVersion: k8ssandra.io/v1alpha1
kind: K8ssandraCluster
metadata:
  namespace: k8ssandra-operator
  name: demo
spec:
  cassandra:
    serverVersion: 4.1.0
    datacenters:
      - metadata:
          name: dc1
        size: 1
        racks:
          - name: rack1
        storageConfig:
          cassandraDataVolumeClaimSpec:
            storageClassName: standard
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 100Mi
```

Simulate a cass-operator upgrade by setting the condition on the `CassandraDatacenter` manually:
```zsh
# Find out the index of the condition in the array. If this is different than 10,
# you need to change it in the next command
kubectl get CassandraDatacenter dc1 -n k8ssandra-operator -o json \
  | jq '.status.conditions | map(.type == "RequiresUpdate") | index(true)'

kubectl patch CassandraDatacenter dc1 -n k8ssandra-operator \
  --subresource status \
  --type='json' \
  -p='[{"op": "replace", "path": "/status/conditions/10/status", "value": "True"}]'
```

Set the annotation on the cluster:
```zsh
kubectl annotate K8ssandraCluster demo -nk8ssandra-operator \
  k8ssandra.io/autoupdate-spec=once
```

Expected: a single `K8ssandraTask` is created.
Actual: a new one is created every ~20s.